core/src/main/java/org/apache/calcite/prepare/PlannerImpl.java

```diff
@@ -362,9 +362,11 @@ private RexBuilder createRexBuilder() {
     return requireNonNull(typeFactory, "typeFactory");
   }
 
+  @SuppressWarnings("deprecation")
   @Override public RelNode transform(int ruleSetIndex, RelTraitSet requiredOutputTraits,
       RelNode rel) {
     ensure(State.STATE_5_CONVERTED);
+    //Remove this when CachingRelMetadataProvider is removed.
```

Comment:
Comment can be removed, it will become evident that method below needs to change if `CachingRelMetadataProvider` is not present.